lib/ceedling/test_includes_extractor.rb

```diff
@@ -65,12 +70,38 @@ def extract_from_file(file)
     return includes.uniq
   end
 
+  def extract_source_include_from_file(file)
+    source_includes = []
+    source_extension = @configurator.extension_source
+
+    contents = @file_wrapper.read(file)
+
+    # remove line comments
+    contents = contents.gsub(/\/\/.*$/, '')
+    # remove block comments
+    contents = contents.gsub(/\/\*.*?\*\//m, '')
+
+    contents.split("\n").each do |line|
+      # look for include statement
+      scan_results = line.scan(/#include\s+\"\s*(.+#{'\\'+source_extension})\s*\"/)
+
+      source_includes << scan_results[0][0] if (scan_results.size > 0)
+
+      # look for TEST_FILE statement
+      scan_results = line.scan(/TEST_FILE\(\s*\"\s*(.+\.\w+)\s*\"\s*\)/)
+
+      source_includes << scan_results[0][0] if (scan_results.size > 0)
```

Comment:
Are you sure, you need this change? With these lines Ceedling `temp_sensor` project failed to compile. I think TEST_FILE() feature is used intentionally to link only external dependencies Check for example issues #295, #113

Reply:
@CezaryGapinski I agree with you. TEST_FILE is actually to add object to be linked and here I will be selecting it to be removed from linker. I will remove it.